controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -916,9 +1059,32 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 			err = k8sClient.Delete(ctx, dtc)
 			Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
 		})
+
+		It("Skip ephemeral env creation for passed Snapshot", func() {
+			var buf bytes.Buffer
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+			updatedSnapshot, _ := gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test pass")
+			Expect(gitops.HaveAppStudioTestsSucceeded(updatedSnapshot)).To(BeTrue())
+			adapter = NewAdapter(updatedSnapshot, hasApp, hasComp, log, loader.NewMockLoader(), k8sClient, ctx)
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   updatedSnapshot,
+				},
+			})
+			result, err := adapter.EnsureCreationOfEphemeralEnvironments()
+			Expect(buf.String()).Should(ContainSubstring("The Snapshot has finished testing."))
+			Expect(!result.CancelRequest).To(BeTrue())
```

Comment:
```suggestion
			Expect(result.CancelRequest).To(BeFalse())
```
Looks less obfuscated.

Reply:
Updated.